src/main/java/org/lushplugins/chatcolorhandler/parsers/custom/SoundParser.java

```diff
@@ -0,0 +1,68 @@
+package org.lushplugins.chatcolorhandler.parsers.custom;
+
+import org.bukkit.Location;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+import org.lushplugins.chatcolorhandler.parsers.Parser;
+import org.lushplugins.chatcolorhandler.parsers.ParserTypes;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SoundParser implements Parser {
+
+    private static final Pattern SOUND_PATTERN =
+            Pattern.compile("<sound:([A-Za-z0-9._-]+)(?::([0-9.]+))?(?::([0-9.]+))?>");
+
+    private static final float DEFAULT_VOLUME = 1.0f;
+    private static final float DEFAULT_PITCH = 1.0f;
+
+    public static final SoundParser INSTANCE = new SoundParser();
+
+    private SoundParser() {}
+
+    @Override
+    public String getType() {
+        return ParserTypes.SOUND;
+    }
+
+    @Override
+    public String parseString(@NotNull String string, @NotNull OutputType outputType) {
+        return string;
+    }
+
+    @Override
+    public String parseString(@NotNull String string, @NotNull OutputType outputType, Player player) {
+        if (!string.contains("<sound:")) {
+            return string;
+        }
+
+        Matcher soundMatcher = SOUND_PATTERN.matcher(string);
+
+        if (!soundMatcher.find()) {
+            return string;
+        }
+
+        soundMatcher.reset();
+
+        while (soundMatcher.find()) {
+            try {
+                String soundName = soundMatcher.group(1);
+                Sound sound = Sound.valueOf(soundName);
```

Comment:
```suggestion
                String sound = soundMatcher.group(1);
```
Changing to just using the string will allow custom sounds from resource packs to be played. This should also allow you to remove the try-catch

Reply:
Im pretty sure the string parameter is *only* for custom sounds.

Reply:
It's not, the clientbound play sound packet only carries a string, the play sound method basically just sends the packet

Reply:
Oh yeah, my bad you're right.

The issue seems to be that the string only accepts lowercase characters (`[a-z0-9/._-]`), while sounds are all uppercase (e.g., `ENTITY_EXPERIENCE_ORB_PICKUP`).

Maybe something like this could work?

````Java
try {
  Sound sound = Sound.valueOf(soundString);
  player.playSound(player, sound, volume, pitch);
} catch (IllegalArgumentException e) {
  player.playSound(player, soundString.toLowerCase(), volume, pitch);
}
````

Reply:
Just `String sound = soundMatcher.group(1).toLowerCase();` would be suitable, the only reason they are uppercase is due to the enum but Mojang's NamespacedKeys are all lower case, the enum basically just converts it to the namespaced key
![image](https://github.com/user-attachments/assets/90842707-47ee-46cf-afed-980d57c314b4)

Reply:
While that works for most sounds, there are also some keys that contain `_`, which is probably going to be an issue. e.g. `BLOCK_COMPOSTER_FILL_SUCCESS("block.composter.fill_success")`

Reply:
That is true, in that case I'd rather just use the actual sound namespaces that mojang uses, so just `String sound = soundMatcher.group(1).toLowerCase();`

Reply:
Yeah, just going with the actual Mojang namespaces sounds like the best and simplest solution.